back/engines/commercial/bulk_import_ideas/app/services/bulk_import_ideas/parsers/pdf/gpt_text_corrector.rb

```diff
@@ -19,29 +23,44 @@ def correctable_idea_fields
     core_fields = multiline_fields.filter_map { _1.code&.to_sym }
     custom_fields = multiline_fields.reject(&:code?).map { _1.key&.to_sym }
 
-    @idea_rows.map do |idea_row|
-      idea_to_correct = idea_row.slice(:id, *core_fields)
-      idea_to_correct[:custom_field_values] = idea_row[:custom_field_values].slice(*custom_fields)
+    return [] if core_fields.blank? && custom_fields.blank?
+
+    @idea_rows.filter_map do |idea_row|
+      idea_to_correct = idea_row.slice(*core_fields)
+      values = idea_row[:custom_field_values]&.slice(*custom_fields)
+      idea_to_correct[:custom_field_values] = values if values.present?
+      next if idea_to_correct.blank?
+
+      idea_to_correct[:id] = idea_row[:id]
       idea_to_correct
     end
   end
 
   def prompt(ideas_to_correct)
     <<~GPT_PROMPT
-      At the end of this message is a JSON array of objects. The fields within these objects are strings that have been extracted through the recognition of handwritten text. But in many cases the texts are recognized innacurately. The lines and some words could be mixed up.
+      At the end of this message, you’ll find a JSON array of objects. The fields within these objects contain texts that have been extracted through the recognition of handwritten text. However, in many cases, inaccuracies may occur. Lines and some words could be mixed up.
+
+
+      Your task is to correct the texts to ensure that each one makes sense. Follow these rules:
+      - Do not remove anything from the original text.
+      - Don't skip words that make sense in the context. Consider rather fixing a typo in a word than removing it.
+      - Maintain the meaning without altering it.
+      - Leave numbers unchanged. E.g., "3" should remain "3", not "three". The same applies to all other digits from 0 to 9.
+      - If a text already makes sense, leave it unchanged.
```

Comment:
We need more data to test, but these concrete instructions seem to improve results pretty consistently.